### PR TITLE
clusterlogging pkg: objectBucketClaim object handler added

### PR DESCRIPTION
**Pull request by elenagerman**

### CODE REVIEW

**Comment by elenagerman** [comment]
@trewest @kononovn @achuzhoy @cdvultur PTAL

**Comment by elenagerman** [comment]
@trewest @kononovn @achuzhoy @cdvultur PTAL

File: pkg/clusterlogging/objectbucketclaim.go [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	noobaav1alpha1 "github.com/kube-object-storage/lib-bucket-provisioner/pkg/apis/objectbucket.io/v1alpha1"
+)
+
+// ObjectBucketClaimBuilder provides struct for the objectBucketClaim object.
+type ObjectBucketClaimBuilder struct {
```
Comment (kononovn) [inline pkg/clusterlogging/objectbucketclaim.go:18]: According to the documentation this is storage object.
https://rook.io/docs/rook/latest-release/Storage-Configuration/Object-Storage-RGW/ceph-object-bucket-claim/
I think this object has to be under storage package.

File: pkg/clusterlogging/objectbucketclaim.go [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+	return builder, err
+}
+
+// Delete removes objectBucketClaim from a cluster.
+func (builder *ObjectBucketClaimBuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting the objectBucketClaim %s in namespace %s",
```
Comment (kononovn) [inline pkg/clusterlogging/objectbucketclaim.go:156]: Deleting the objectBucketClaim %s from namespace %s",

File: pkg/clusterlogging/objectbucketclaim_test.go [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+		result := testBuilder.WithStorageClassName(testCase.testStorageClassName)
+		assert.Equal(t, testCase.expectedError, result.errorMsg)
+
+		if testCase.expectedError == "" {
+			assert.NotNil(t, result)
+			assert.Equal(t, testCase.testStorageClassName, result.Definition.Spec.StorageClassName)
+		}
+	}
+}
+
```
Comment (kononovn) [inline pkg/clusterlogging/objectbucketclaim_test.go:330]: 1 test case missing. Please add TestObjectBucketClaimWithGenerateBucketName

**Review by kononovn — CHANGES_REQUESTED** [review]
Looks good just few comments to fix.

**Review by kononovn — DISMISSED** [review]
/lgtm

File: pkg/storage/objectbucketclaim.go [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+	if err != nil {
+		glog.V(100).Infof(
+			msg.FailToUpdateError("objectBucketClaim", builder.Definition.Name, builder.Definition.Namespace))
+
+		return nil, err
+	}
+
+	if err == nil {
+		builder.Object = builder.Definition
+	}
```
Comment (trewest) [inline pkg/storage/objectbucketclaim.go:215]:
```suggestion
	builder.Object = builder.Definition
```

**Pull request merged**